Sources/Result/ValidationResult.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+
+
+/// Validation result type.
+///
+/// - success: Validation was succesfull
+/// - failure: Validation failed. Contains `ValidationResult.Summary` that details the cause for failure
+public enum ValidationResult {
+    case success
+    case failure(Summary)
+}
+
+public extension ValidationResult {
+    
+    internal init(summary: Summary) {
+        
+        if summary.errors.isEmpty {
+            self = .success
+        }
+        else {
+            self = .failure(summary)
+        }
+    }
+    
+    /**
+     `true` if the validation result is valid, `false` otherwise.
+     */
+    var isValid: Bool {
```

Comment:
Taking into account the renaming of `success`/ `failure`, in what measure do you think it makes sense to find more appropriate names for these two properties?

```suggestion
    var isSuccessful: Bool {
```

Reply:
👍